### PR TITLE
chore: remove OsRng

### PR DESCRIPTION
**Pull request by iancoleman**
`// TODO: OsRng needs to be removed + replaced to follow no-std.`

### CODE REVIEW

**Comment by iancoleman** [comment]
Worth adding for completeness, this change was motivated by trying to compile this module to wasm which requires OsRng to be removed.

File: src/lib.rs [hunk 1]
```diff
@@ -128,11 +126,9 @@ impl XorName {
     }
 
     /// Generate a random XorName
-    pub fn random() -> Self {
+    pub fn random<T: Rng>(rng: &mut T) -> Self {
```
Comment (dan-da) [inline src/lib.rs:129]: This is clearly a breaking change.

We could possibly keep random() unchanged and add another random_rng(), or possibly sample() method that takes Rng param.

I guess an argument can be made that it is correct/better to force callers to pass in the Rng.  So I'm not saying this is wrong, just inquiring about the rationale.

File: Cargo.toml [hunk 2]
```diff
@@ -27,15 +27,13 @@ rand_core = "0.5.1"
...
 bincode = "1.2.1"
-eyre = "0.6.5"
 
   [dev-dependencies.arrayvec]
   version = "~0.5.1"
   default-features = false
 
   [dev-dependencies.rand]
-  version = "~0.7.3"
+  version = "~0.8.5"
```
Comment (dan-da) [inline Cargo.toml:37]: yay!

**Review by dan-da — COMMENTED** [review]
lgtm, with caveat that probably someone working on core SN libs should review it because I'm unsure how much work it would be to change all the calls to xor_name::random().

File: src/lib.rs (hunk 1 above)
Comment (iancoleman) [reply src/lib.rs:129]: A bit of history of the `random()` function to save others digging

https://github.com/maidsafe/xor_name/commit/20b9ad5aef133c4fdc019cc901123d8d79849bde - feat(random): Add random xorname func
https://github.com/maidsafe/xor_name/commit/0a8325226ac0cbc5b3e7b315faa64654f78d4d24 - fix(rng): Use OsRng for security

Both in in https://github.com/maidsafe/xor_name/pull/41

**Review by joshuef — CHANGES_REQUESTED** [review]
@iancoleman [remove OsRng](https://github.com/maidsafe/xor_name/pull/71/commits/7caab944fbcef7ab0ceaf53600e777c1c28d2268) will need a `BREAKING CHANGE:` in the commit text, (or `!` on the subject) to properly version this.

**Review by dan-da — APPROVED** [review]
There is a `BREAKING CHANGE:` in the commit message now, so I'm going to approve.

**Pull request merged**